### ISSUE TITLE
OMP rendering generic linux icon instead of ubuntu icon in WSL Ubuntu 20.04

### ISSUE THREAD

**Issue by devhawk** [JanDeDobbeleer/oh-my-posh#1766]
### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### What happened?

Screenshot: 
![image](https://user-images.githubusercontent.com/8965/154342828-c2e984f2-2c69-402e-a411-7c93c0b6398b.png)

### Theme

Relevant Segments: 

```json
        {
          "background": "#4082ce",
          "background_templates": [
            "{{if contains \"Ubuntu\" .Env.WSL_DISTRO_NAME}}#dd4814{{end}}",
            "{{if contains \"Debian\" .Env.WSL_DISTRO_NAME}}#D70A53{{end}}",
          ],
          "foreground": "#ffffff",
          "leading_diamond": "\ue0b6",
          "properties": {
            "template": "{{.Icon}}",
            "windows": "\uf17a"
          },
          "style": "diamond",
          "type": "os"
        },
        {
          "background": "#4082ce",
          "background_templates": [
            "{{if contains \"Ubuntu\" .Env.WSL_DISTRO_NAME}}#dd4814{{end}}",
            "{{if contains \"Debian\" .Env.WSL_DISTRO_NAME}}#D70A53{{end}}",
          ],
          "foreground": "#ffffff",
          "properties": {
            "template": " {{ .Env.WSL_DISTRO_NAME }}"
          },
          "style": "diamond",
          "trailing_diamond": "\ue0b4",
          "type": "text"
        },
```

### What OS are you seeing the problem on?

Linux

### Which shell are you using?

bash

### Log output

```shell
[1mVersion:[0m 7.18.0

[1mSegments:[0m

ConsoleTitle(true) -   0 ms - ]0;~ / Ubuntu-20.04
path(true)     -   1 ms - [38;2;49;57;127m[0m[48;2;49;57;127m[38;2;255;255;255m ~ [0m[38;2;49;57;127m[0m
battery(false) -   0 ms - 
git(false)     -   0 ms - 
node(false)    -   0 ms - 
dotnet(false)  -   0 ms - 
rust(false)    -   0 ms - 
python(false)  -   0 ms - 
go(false)      -   0 ms - 
os(true)       -   0 ms - [38;2;221;72;20m[0m[48;2;221;72;20m[38;2;255;255;255m[0m
text(true)     -   0 ms - [48;2;221;72;20m[38;2;255;255;255m Ubuntu-20.04[0m[38;2;221;72;20m[0m
time(true)     -   0 ms - [39m 11:42 am [0m
root(false)    -   0 ms - 
exit(false)    -   0 ms - 
path(true)     -   1 ms - [38;2;49;57;127;49m[7m[0m[48;2;49;57;127m[38;2;255;255;255m ~ [0m[38;2;49;57;127m[0m

[1mRun duration:[0m 4.6599ms

[1mCache path:[0m /home/harry/.cache/oh-my-posh

[1mLogs:[0m

2022/02/16 11:42:28 Args duration: 200ns, args: 
2022/02/16 11:42:28 Args duration: 100ns, args: 
2022/02/16 11:42:28 Args duration: 200ns, args: 
2022/02/16 11:42:28 Shell duration: 200ns, args: 
2022/02/16 11:42:28 debug: Getenv

2022/02/16 11:42:28 Getenv duration: 6.2µs, args: OMP_CACHE_DISABLED
2022/02/16 11:42:28 Root duration: 400ns, args: 
2022/02/16 11:42:28 Shell duration: 200ns, args: 
2022/02/16 11:42:28 ErrorCode duration: 100ns, args: 
2022/02/16 11:42:28 debug: Getenv
Ubuntu-20.04
2022/02/16 11:42:28 Getenv duration: 1.5µs, args: WSL_DISTRO_NAME
2022/02/16 11:42:28 IsWsl duration: 3.8µs, args: 
2022/02/16 11:42:28 debug: Pwd
/home/harry
2022/02/16 11:42:28 Pwd duration: 9.7µs, args: 
2022/02/16 11:42:28 PathSeperator duration: 200ns, args: 
2022/02/16 11:42:28 PathSeperator duration: 0s, args: 
2022/02/16 11:42:28 User duration: 100ns, args: 
2022/02/16 11:42:28 Host duration: 1.1µs, args: 
2022/02/16 11:42:28 GOOS duration: 200ns, args: 
2022/02/16 11:42:28 TemplateCache duration: 47.2µs, args: 
2022/02/16 11:42:28 debug: Pwd
/home/harry
2022/02/16 11:42:28 Pwd duration: 1µs, args: 
2022/02/16 11:42:28 debug: Pwd
/home/harry
2022/02/16 11:42:28 Pwd duration: 1.3µs, args: 
2022/02/16 11:42:28 Args duration: 100ns, args: 
2022/02/16 11:42:28 debug: Pwd
/home/harry
2022/02/16 11:42:28 Pwd duration: 900ns, args: 
2022/02/16 11:42:28 GOOS duration: 0s, args: 
2022/02/16 11:42:28 GOOS duration: 0s, args: 
2022/02/16 11:42:28 PathSeperator duration: 100ns, args: 
2022/02/16 11:42:28 GOOS duration: 0s, args: 
2022/02/16 11:42:28 debug: Getenv
Ubuntu-20.04
2022/02/16 11:42:28 Getenv duration: 900ns, args: WSL_DISTRO_NAME
2022/02/16 11:42:28 IsWsl duration: 2.7µs, args: 
2022/02/16 11:42:28 debug: RunCommand
//wsl$/Ubuntu-20.04/home/harry
2022/02/16 11:42:28 RunCommand duration: 1.1154ms, args: wslpath -m /home/harry
2022/02/16 11:42:28 StackCount duration: 100ns, args: 
2022/02/16 11:42:28 TemplateCache duration: 100ns, args: 
2022/02/16 11:42:28 debug: Pwd
/home/harry
2022/02/16 11:42:28 Pwd duration: 1.3µs, args: 
2022/02/16 11:42:28 debug: Getenv
Ubuntu-20.04
2022/02/16 11:42:28 Getenv duration: 900ns, args: WSL_DISTRO_NAME
2022/02/16 11:42:28 IsWsl duration: 2.8µs, args: 
2022/02/16 11:42:28 debug: Getenv
Ubuntu-20.04
2022/02/16 11:42:28 Getenv duration: 700ns, args: WSL_DISTRO_NAME
2022/02/16 11:42:28 IsWsl duration: 2.5µs, args: 
2022/02/16 11:42:28 FileContent duration: 29.7µs, args: /proc/sys/kernel/osrelease
2022/02/16 11:42:28 IsWsl2 duration: 37.9µs, args: 
2022/02/16 11:42:28 debug: Pwd
/home/harry
2022/02/16 11:42:28 Pwd duration: 900ns, args: 
2022/02/16 11:42:28 debug: Getenv
Ubuntu-20.04
2022/02/16 11:42:28 Getenv duration: 900ns, args: WSL_DISTRO_NAME
2022/02/16 11:42:28 IsWsl duration: 2.3µs, args: 
2022/02/16 11:42:28 debug: Pwd
/home/harry
2022/02/16 11:42:28 Pwd duration: 600ns, args: 
2022/02/16 11:42:28 GOOS duration: 0s, args: 
2022/02/16 11:42:28 HasCommand duration: 11.2µs, args: git
2022/02/16 11:42:28 debug: Pwd
/home/harry
2022/02/16 11:42:28 Pwd duration: 800ns, args: 
2022/02/16 11:42:28 error: HasParentFilePath
stat /.git: no such file or directory
2022/02/16 11:42:28 HasParentFilePath duration: 6.6µs, args: .git
2022/02/16 11:42:28 debug: Pwd
/home/harry
2022/02/16 11:42:28 Pwd duration: 800ns, args: 
2022/02/16 11:42:28 debug: Pwd
/home/harry
2022/02/16 11:42:28 Pwd duration: 700ns, args: 
2022/02/16 11:42:28 debug: Pwd
/home/harry
2022/02/16 11:42:28 Pwd duration: 700ns, args: 
2022/02/16 11:42:28 debug: Pwd
/home/harry
2022/02/16 11:42:28 Pwd duration: 800ns, args: 
2022/02/16 11:42:28 debug: Pwd
/home/harry
2022/02/16 11:42:28 Pwd duration: 800ns, args: 
2022/02/16 11:42:28 debug: Pwd
/home/harry
2022/02/16 11:42:28 Pwd duration: 700ns, args: 
2022/02/16 11:42:28 debug: Pwd
/home/harry
2022/02/16 11:42:28 Pwd duration: 700ns, args: 
2022/02/16 11:42:28 debug: Pwd
/home/harry
2022/02/16 11:42:28 Pwd duration: 700ns, args: 
2022/02/16 11:42:28 debug: Getenv

2022/02/16 11:42:28 Getenv duration: 900ns, args: VIRTUAL_ENV
2022/02/16 11:42:28 debug: Getenv

2022/02/16 11:42:28 Getenv duration: 800ns, args: CONDA_ENV_PATH
2022/02/16 11:42:28 debug: Getenv

2022/02/16 11:42:28 Getenv duration: 700ns, args: CONDA_DEFAULT_ENV
2022/02/16 11:42:28 debug: Getenv

2022/02/16 11:42:28 Getenv duration: 900ns, args: PYENV_VERSION
2022/02/16 11:42:28 debug: Pwd
/home/harry
2022/02/16 11:42:28 Pwd duration: 800ns, args: 
2022/02/16 11:42:28 PathSeperator duration: 0s, args: 
2022/02/16 11:42:28 HasFiles duration: 53.3µs, args: *.py
2022/02/16 11:42:28 debug: Pwd
/home/harry
2022/02/16 11:42:28 Pwd duration: 800ns, args: 
2022/02/16 11:42:28 PathSeperator duration: 0s, args: 
2022/02/16 11:42:28 HasFiles duration: 22.3µs, args: *.ipynb
2022/02/16 11:42:28 debug: Pwd
/home/harry
2022/02/16 11:42:28 Pwd duration: 700ns, args: 
2022/02/16 11:42:28 PathSeperator duration: 0s, args: 
2022/02/16 11:42:28 HasFiles duration: 4.4µs, args: pyproject.toml
2022/02/16 11:42:28 debug: Pwd
/home/harry
2022/02/16 11:42:28 Pwd duration: 600ns, args: 
2022/02/16 11:42:28 PathSeperator duration: 100ns, args: 
2022/02/16 11:42:28 HasFiles duration: 3.7µs, args: venv.bak
2022/02/16 11:42:28 debug: Pwd
/home/harry
2022/02/16 11:42:28 Pwd duration: 700ns, args: 
2022/02/16 11:42:28 PathSeperator duration: 0s, args: 
2022/02/16 11:42:28 HasFiles duration: 4.5µs, args: venv
2022/02/16 11:42:28 debug: Pwd
/home/harry
2022/02/16 11:42:28 Pwd duration: 1.6µs, args: 
2022/02/16 11:42:28 PathSeperator duration: 100ns, args: 
2022/02/16 11:42:28 HasFiles duration: 5.4µs, args: .venv
2022/02/16 11:42:28 debug: Pwd
/home/harry
2022/02/16 11:42:28 Pwd duration: 800ns, args: 
2022/02/16 11:42:28 debug: Pwd
/home/harry
2022/02/16 11:42:28 Pwd duration: 700ns, args: 
2022/02/16 11:42:28 debug: Pwd
/home/harry
2022/02/16 11:42:28 Pwd duration: 800ns, args: 
2022/02/16 11:42:28 GOOS duration: 0s, args: 
2022/02/16 11:42:28 TemplateCache duration: 100ns, args: 
2022/02/16 11:42:28 TemplateCache duration: 200ns, args: 
2022/02/16 11:42:28 TemplateCache duration: 200ns, args: 
2022/02/16 11:42:28 debug: Pwd
/home/harry
2022/02/16 11:42:28 Pwd duration: 1.4µs, args: 
2022/02/16 11:42:28 TemplateCache duration: 200ns, args: 
2022/02/16 11:42:28 TemplateCache duration: 100ns, args: 
2022/02/16 11:42:28 TemplateCache duration: 100ns, args: 
2022/02/16 11:42:28 debug: Pwd
/home/harry
2022/02/16 11:42:28 Pwd duration: 19.4µs, args: 
2022/02/16 11:42:28 TemplateCache duration: 200ns, args: 
2022/02/16 11:42:28 debug: Pwd
/home/harry
2022/02/16 11:42:28 Pwd duration: 1µs, args: 
2022/02/16 11:42:28 Root duration: 500ns, args: 
2022/02/16 11:42:28 debug: Pwd
/home/harry
2022/02/16 11:42:28 Pwd duration: 800ns, args: 
2022/02/16 11:42:28 ErrorCode duration: 100ns, args: 
2022/02/16 11:42:28 ErrorCode duration: 0s, args: 
2022/02/16 11:42:28 debug: Pwd
/home/harry
2022/02/16 11:42:28 Pwd duration: 1.1µs, args: 
2022/02/16 11:42:28 debug: Pwd
/home/harry
2022/02/16 11:42:28 Pwd duration: 1.1µs, args: 
2022/02/16 11:42:28 Args duration: 0s, args: 
2022/02/16 11:42:28 debug: Pwd
/home/harry
2022/02/16 11:42:28 Pwd duration: 800ns, args: 
2022/02/16 11:42:28 GOOS duration: 100ns, args: 
2022/02/16 11:42:28 GOOS duration: 0s, args: 
2022/02/16 11:42:28 PathSeperator duration: 100ns, args: 
2022/02/16 11:42:28 PathSeperator duration: 0s, args: 
2022/02/16 11:42:28 PathSeperator duration: 100ns, args: 
2022/02/16 11:42:28 GOOS duration: 100ns, args: 
2022/02/16 11:42:28 debug: Getenv
Ubuntu-20.04
2022/02/16 11:42:28 Getenv duration: 900ns, args: WSL_DISTRO_NAME
2022/02/16 11:42:28 IsWsl duration: 2.5µs, args: 
2022/02/16 11:42:28 debug: RunCommand
//wsl$/Ubuntu-20.04/home/harry
2022/02/16 11:42:28 RunCommand duration: 957.9µs, args: wslpath -m /home/harry
2022/02/16 11:42:28 StackCount duration: 100ns, args: 
2022/02/16 11:42:28 TemplateCache duration: 100ns, args: 
2022/02/16 11:42:28 debug: Getenv

2022/02/16 11:42:28 Getenv duration: 1.5µs, args: XDG_CACHE_HOME
2022/02/16 11:42:28 CachePath duration: 8.9µs, args:
```

**Comment by JanDeDobbeleer** [comment]
@devhawk pretty sure that's because the `WSL_DISTRO_NAME` variable contains `Ubuntu-20.04` where we expect a simple `Ubuntu`.

**Comment by lewis-yeung** [comment]
This bug persists in the latest version (7.42.0). I noticed a fact, after emptying the contents of the cache file `$POSH_PATH/omp.cache` and pressing Enter in my terminal, the OMP engine rendered an expected icon of the WSL distro (which is in my case, Ubuntu) for my prompt. And then I found a new value of "ubuntu-20.04" was spawned in the cache file:

```json
{
    "environment_platform": {
        "value": "ubuntu",
        "timestamp": 1647627200,
        "ttl": -1
    }
}
```

After that, it will always render a generic Linux icon in subsequent prompts unless I delete the entry of `environment_platform` again.

After looking into the code, I found that the distro name was not properly stored after being split from the value of the environment variable `WSL_DISTRO_NAME`, instead the original value of `WSL_DISTRO_NAME` was stored. Since the value matches none of the preset distro names, a generic Linux icon is used as fallback. Besides, the value is stored with a TTL of -1, which means "never expire", so it will not be updated.

I fixed the code, deleted the cache file and tested the newly built OMP executable, it works well.

A PR (#1944) has been submitted.
